R/zchunk_L226.en_distribution.R

```diff
@@ -65,172 +69,358 @@ module_energy_L226.en_distribution_DISABLED <- function(command, ...) {
+    A26.globaltech_shrwt %>%
+      gather(year, share.weight, -supplysector, -subsector, -technology) %>%
+      complete(nesting(supplysector, subsector, technology), year = c(year, BASE_YEARS, FUTURE_YEARS)) %>%
+      arrange(supplysector, year) %>%
+      group_by(supplysector) %>%
+      mutate (share.weight = approx_fun(as.numeric(year), share.weight)) %>%
+      ungroup() %>%
+      filter(year %in% c(BASE_YEARS, FUTURE_YEARS)) %>%
+      #Assign the columns "sector.name" and "subsector.name", consistent with the location info of a global technology
+      rename(sector.name = supplysector, subsector.name = subsector) ->
+      L226.GlobalTechShrwt_en
+
+    #2d. Calibration and region-specific data
+    # Electricity ownuse IO coefs - filter down to the base years and append region IDs
+    L126.IO_R_elecownuse_F_Yh %>%
+      ungroup() %>% #this data apparently came into this chunk grouped, which was preventing deletion of grouped columns
+      filter(year %in% c(BASE_YEARS)) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      left_join(calibrated_techs, by = c("sector", "fuel")) %>%
+      select(-calibration, -secondary.output) ->
+      L226.IO_R_elecownuse_F_Yh
+
+    #repeat final year's ownuse ratio into future years and append future years to base years (could perhaps be tied to industrial CHP...but also AUTOELEC)
+    L226.IO_R_elecownuse_F_Yh %>%
+      filter(year == max(BASE_YEARS)) %>%
+      repeat_add_columns(tibble("year" = FUTURE_YEARS)) %>%
+      select(-year.x) %>%
+      rename(year = year.y) %>%
+      bind_rows(L226.IO_R_elecownuse_F_Yh, .) ->
+      L226.IO_R_elecownuse_F_Y
+
+    #rename columns and round coefficients - L226.StubTechCoef_elecownuse: calibrated coefficients on electricity net ownuse
+    L226.IO_R_elecownuse_F_Y %>%
+      rename(coefficient = value, stub.technology = technology) %>%
+      mutate(market.name = region, coefficient = round(coefficient, DIGITS_COEFFICIENT)) %>%
+      select(-GCAM_region_ID, -sector, -fuel) ->
+      L226.StubTechCoef_elecownuse
+
+    #Filter electricity transmission and distribution input-output ratio historical data down to base years (#this works now but may need optional interpolation if the assumptions file changes not to include base model years)
+    L126.IO_R_electd_F_Yh %>%
+      filter(year %in% BASE_YEARS) ->
+      L226.IO_R_electd_F_Yh
+
+    # Copy final base year value to future periods
+    L226.IO_R_electd_F_Yh %>%
+      filter(year == max(BASE_YEARS)) %>%
+      repeat_add_columns(tibble("year" = FUTURE_YEARS)) %>%
+      select(-year.x) %>%
+      rename(year = year.y) ->
+      L226.IO_R_electd_F_Yfut
+
+    # append assumed techchange value and calculate decrease in future energy use
+    L226.IO_R_electd_F_Yfut %>%
+      left_join(A_regions, by = "GCAM_region_ID") %>%
```

Comment:
Why not `left_join_error_no_match`? Are there region IDs not present in `A_regions`?

Reply:
no, but there's one entry in biomassoil_Tech that's NA and so triggers the error when it copies it (at least I assume that's what's happening).